presto-orc/src/main/java/io/prestosql/orc/writer/TimestampColumnWriter.java

```diff
@@ -261,4 +317,108 @@ public void reset()
         rowGroupColumnStatistics.clear();
         statisticsBuilder = statisticsBuilderSupplier.get();
     }
+
+    private void writeMillis(long millis)
+    {
+        long seconds = millis / MILLISECONDS_PER_SECOND;
```

Comment:
Is `millis` an epoch-based value? This needs to be floorDiv, then.

Reply:
This is a flaw in the ORC encoding, as explained in the comment at the top of this class. I'll add a separate comment to each of the instances so obvious this is not a bug.